### ISSUE TITLE
Azure MySQL connection failed, even { encrypt: true }

### ISSUE THREAD

**Issue by jyunhan** [sequelize/sequelize#7669]
Hi, I face a strange situation,
I connect to local DB or DB at Azure VM are good,
but when we purchase a Azure DB solution and trying to change our host
we found an error:

**_Unhandled rejection SequelizeConnectionError: MySQL server is requesting the old and insecure pre-4.1 auth mechanism.Upgrade the user password or
 use the {insecureAuth: true} option._**

I look related issue and suggestion
to adjust connection like:

_**new Sequelize(properties.get('database.name'), properties.get('database.user'), properties.get('database.password'), {
        host: configMap.host,
        dialect: configMap.db_brand,
        pool: {
			max: 5,
			min: 0,
			idle: 10000
		}
		, logging: false
		, timezone: '+08:00'
		, dialectOptions: {
			encrypt: true ( or insecureAuth: true)
		}
    });**_

but the error still appear.
even I switch dialectOptions object inside to insecureAuth: true
the new error statement occurs:
**_Unhandled rejection SequelizeConnectionError: read ECONNRESET_**

does I lost any adjustment..?

sequelize version: 3.27.0

**Comment by jyunhan** [comment]
We'v ask Microsoft Taiwan this issue, cos' we found It might not sequelize problem.
And the response is, Azure's mysql solution is not support all feature completely so far.
so, we change our server env. and everything is getting smoother.

**Comment by kirikintha** [comment]
Hello - @jyunhan - can you please elaborate on what "so, we change our server env." means? 

Are you still using the Preview version of Azure MySQL?
If not, what did you move to?
If you are still using the Azure MySQL (Preview), can you please share what you did to make it work? IE: connection settings, etc.

I realize MySQL is in preview - so anyone else who finds this can understand why it is not a Sequelize problem. Thank you for helping me out with your explanation as well.

**Comment by jyunhan** [comment]
sure, here is the scenario.

We purchased Azure's mysql solution and an cloud service to migrating our service on that, include our database.
But database was created failed at all times on Azure mysql solution platform,
and command said that Sequelize.js occurred some problem. 
I took a while to trace code and confirm migrating procedure were fine.

So, I thought It maybe happened on new platform.
Therefore, I installed mysql on an other cloud service and connected to db server manually, then we got connected error.

According to response from Microsoft customer service, and their tech-member.
We decided to purchase an other cloud service and build mysql server by myself, to instead Azure's mysql solution platform.

disadvantages :
